.buildkite/steps/tests.sh

```diff
@@ -1,13 +1,13 @@
 #!/bin/bash
-set -euo pipefail
+set -eo pipefail
 
 go version
 echo arch is "$(uname -m)"
 
 go install gotest.tools/gotestsum@v1.8.0
 
 echo '+++ Running tests'
-gotestsum --junitfile "junit-${OSTYPE}.xml" -- -count=1 -failfast "$@" ./...
+gotestsum --junitfile "junit-${BUILDKITE_STEP_KEY}.xml" -- -count=1 -failfast "$@" ./...
 
 echo '+++ Running integration tests for git-mirrors experiment'
-TEST_EXPERIMENT=git-mirrors gotestsum --junitfile "junit-${OSTYPE}-git-mirrors.xml" -- -count=1 -failfast "$@" ./bootstrap/integration
+TEST_EXPERIMENT=git-mirrors gotestsum --junitfile "junit-${BUILDKITE_STEP_KEY}-git-mirrors.xml" -- -count=1 -failfast "$@" ./bootstrap/integration
```

Comment:
In general, if you expect one variable to not be set, I think `set +u` is too broad. you should use a pattern like
```bash
JUNIT_FILE_KEY=${BUILDKITE_STEP_KEY:-$OSTYPE}
```
instead

Reply:
You could use BUILDKITE_JOB_ID. That's guaranteed to exist and be unique. I'm unsure why BUILDKITE_STEP_KEY wasn't defined in that example though...

Reply:
Yeah, i'm not sure why that happened either... regardless, i've updated it to keep `set -u` and use `BUILDKITE_JOB_ID` instead